presto-native-execution/presto_cpp/main/common/Configs.h

```diff
@@ -58,8 +65,9 @@ class ConfigBase {
     return config_->get(propertyName);
   }
 
-  const std::unordered_map<std::string, std::string>& values() const {
-    return config_->values();
+  /// Returns copy of the config values map.
+  std::unordered_map<std::string, std::string> values() const {
```

Comment:
why this change?

Reply:
Because underneath can be a mutable config and we cannot return reference to the map which then can be looped through while someone changing it from another thread.
This method is only called during startup.

Reply:
But there is no protection either when you are making the copy in valuesCopy() method. Do we want to add a lock if we want it to be safe?

Reply:
The lock is inside config_->valuesCopy();